packages/modular-scripts/src/utils/gitActions.ts

```diff
@@ -17,3 +18,78 @@ export function stashChanges(): void {
   execa.sync('git', ['stash', '-u']);
   throw new Error('Failed to perform action cleanly. Stashing git changes...');
 }
+
+function getGitLocalDefaultBranch(): string {
+  try {
+    // this config may be set to a different default branch (e.g. main)
+    const result = execa.sync('git', ['config', 'init.defaultBranch']);
+    return stripAnsi(result.stdout);
+  } catch (_err) {
+    // if there init.defaultBranch hasn't been set, the default is master
+    return 'master';
```

Comment:
`init.defaultBranch` has to be explicitly set locally. That config is only present if someone sets it to main or something else on their machine, otherwise the config does not exist and the default is master. This handles the case in which someone has not set it to main and their default branch is master for new local repos

Reply:
Sure - but I think we should follow convention here - see https://www.zdnet.com/article/github-to-replace-master-with-main-starting-next-month/

Reply:
That's only if you are creating a new repo in github. If you do git init -y, it will use your git.initDefaultBranch, which was created to allow for local repos to be initialized with a `main` default branch. Until git itself sets the automatic default branch to main without setting the git.initDefaultBranch config, we have to cater to `master` being the default branch for local repo set up. I agree that we want projects to follow convention that this is to address people who have not done so.

Reply:
Hmm - I personally disagree and think we should at least push people in the right direction, but I agree with your argument. This is likely a unlikely code branch so I'm happy to leave it for the time being.